Mathlib/Algebra/BigOperators/Finsupp.lean

```diff
@@ -59,107 +60,118 @@ variable [Zero M] [Zero M'] [CommMonoid N]
+  refine Finset.prod_subset hs fun x hxs hx => h x hxs ▸ (congr_arg (g x) ?_)
+  exact not_mem_support_iff.1 hx
 #align finsupp.prod_of_support_subset Finsupp.prod_of_support_subset
 #align finsupp.sum_of_support_subset Finsupp.sum_of_support_subset
 
 @[to_additive]
 theorem prod_fintype [Fintype α] (f : α →₀ M) (g : α → M → N) (h : ∀ i, g i 0 = 1) :
-    f.Prod g = ∏ i, g i (f i) :=
+    f.prod g = ∏ i, g i (f i) :=
   f.prod_of_support_subset (subset_univ _) g fun x _ => h x
 #align finsupp.prod_fintype Finsupp.prod_fintype
 #align finsupp.sum_fintype Finsupp.sum_fintype
 
-@[simp, to_additive]
+@[to_additive (attr := simp)]
 theorem prod_single_index {a : α} {b : M} {h : α → M → N} (h_zero : h a 0 = 1) :
-    (single a b).Prod h = h a b :=
+    (single a b).prod h = h a b :=
   calc
-    (single a b).Prod h = ∏ x in {a}, h x (single a b x) :=
+    (single a b).prod h = ∏ x in {a}, h x (single a b x) :=
       prod_of_support_subset _ support_single_subset h fun x hx =>
         (mem_singleton.1 hx).symm ▸ h_zero
     _ = h a b := by simp
-    
+
 #align finsupp.prod_single_index Finsupp.prod_single_index
 #align finsupp.sum_single_index Finsupp.sum_single_index
 
 @[to_additive]
 theorem prod_mapRange_index {f : M → M'} {hf : f 0 = 0} {g : α →₀ M} {h : α → M' → N}
-    (h0 : ∀ a, h a 0 = 1) : (mapRange f hf g).Prod h = g.Prod fun a b => h a (f b) :=
+    (h0 : ∀ a, h a 0 = 1) : (mapRange f hf g).prod h = g.prod fun a b => h a (f b) :=
   Finset.prod_subset support_mapRange fun _ _ H => by rw [not_mem_support_iff.1 H, h0]
 #align finsupp.prod_map_range_index Finsupp.prod_mapRange_index
 #align finsupp.sum_map_range_index Finsupp.sum_mapRange_index
 
-@[simp, to_additive]
-theorem prod_zero_index {h : α → M → N} : (0 : α →₀ M).Prod h = 1 :=
+@[to_additive (attr := simp)]
+theorem prod_zero_index {h : α → M → N} : (0 : α →₀ M).prod h = 1 :=
   rfl
 #align finsupp.prod_zero_index Finsupp.prod_zero_index
 #align finsupp.sum_zero_index Finsupp.sum_zero_index
 
 @[to_additive]
 theorem prod_comm (f : α →₀ M) (g : β →₀ M') (h : α → M → β → M' → N) :
-    (f.Prod fun x v => g.Prod fun x' v' => h x v x' v') =
-      g.Prod fun x' v' => f.Prod fun x v => h x v x' v' :=
+    (f.prod fun x v => g.prod fun x' v' => h x v x' v') =
+      g.prod fun x' v' => f.prod fun x v => h x v x' v' :=
   Finset.prod_comm
 #align finsupp.prod_comm Finsupp.prod_comm
 #align finsupp.sum_comm Finsupp.sum_comm
 
-@[simp, to_additive]
+@[to_additive (attr := simp)]
 theorem prod_ite_eq [DecidableEq α] (f : α →₀ M) (a : α) (b : α → M → N) :
-    (f.Prod fun x v => ite (a = x) (b x v) 1) = ite (a ∈ f.support) (b a (f a)) 1 := by
+    (f.prod fun x v => ite (a = x) (b x v) 1) = ite (a ∈ f.support) (b a (f a)) 1 := by
   dsimp [Finsupp.prod]
   rw [f.support.prod_ite_eq]
 #align finsupp.prod_ite_eq Finsupp.prod_ite_eq
 #align finsupp.sum_ite_eq Finsupp.sum_ite_eq
 
-@[simp]
+-- Porting note: simpnf linter; Left-hand side simplifies from
```

Comment:
Please add a new theorem `sum_ite_self_eq_aux` that prove that the simplified LHS is equal to the RHS and tag it with `simp`. That way we maintain the same simp-set as mathlib 3.